packages/plugin-aws-lambda/src/index.js

```diff
@@ -15,6 +15,23 @@ const BugsnagPluginAwsLambda = {
       appDurationPlugin.reset()
     }
 
+    // AWS add a default unhandledRejection listener that forcefully exits the
+    // process. This breaks reporting of unhandled rejections, so we have to
+    // remove all existing listeners and call them after we handle the rejection
+    if (client._config.autoDetectErrors && client._config.enabledErrorTypes.unhandledRejections) {
+      const listeners = process.listeners('unhandledRejection')
+      process.removeAllListeners('unhandledRejection')
+
+      const BugsnagPluginUnhandledRejection = require('@bugsnag/plugin-node-unhandled-rejection')
```

Comment:
Couple of issues here.

Modules/duplication:
- `require`s should really be at the top of the file
- Since this is bundled, the fact it is low in the file doesn't defer it being loaded (though it does defer the execution of its module scope) and it will result in it being present twice in the app – once bundled in `@bugsnag/node` and once in `@bugsnag/plugin-aws-lambda`

Double execution
- I'm pretty sure the logic here means that the plugin gets executed twice? The second time it won't be around for long enough to actually send the report due to the AWS callback forcefully executing, but there might be duplicate logs?
- The plugin actually has a (non-standard) `destroy()` method that you could call before taking a snapshot of the process `unhandledError` listeners, however, you can't get a reference to the plugin due to the module duplication mentioned above.

Reply:
How can we avoid having it present twice?